language/module.md

```diff
@@ -42,9 +42,15 @@ already explains what a module is and what it can contain.
 | `/examples/`          | Example Puppet snippets that explain how to use the module. They can be used within acceptance tests |
 | `/facts.d/`           | [External facts](https://www.puppet.com/docs/puppet/latest/external_facts.html) that are synced via [pluginsync](https://www.puppet.com/docs/puppet/latest/plugins_in_modules.html) |
 | `/lib/facter/`        | MAY contain custom facts |
-| `/lib/puppet/type/`   | Custom types |
-| `/lib/puppet/provider/` | Custom provider for one or multiple types |
+| `/lib/puppet/type/`   | Custom Resource types |
+| `/lib/puppet/provider/` | Custom provider for one or multiple Resource types |
 | `/lib/puppet/functions/` | Modern functions in Ruby for the new API |
+| `/lib/puppet/datatypes/` | Custom Puppet Data types |
```

Comment:
I think you're thinking of `/types` like https://github.com/puppetlabs/puppetlabs-stdlib/blob/main/types/absolutepath.pp?

Reply:
ah this is answered in https://github.com/puppetlabs/puppet-specifications/pull/158#discussion_r1577858919